src/transformers/models/ibert/modeling_ibert.py

```diff
@@ -0,0 +1,1418 @@
+        super().__init__()
+        self.quant_mode = config.quant_mode
+        self.act_bit = 8
+        self.weight_bit = 8
+        self.bias_bit = 32
+        self.ln_input_bit = 22
+        self.ln_output_bit = 32
+
+        self.dense = QuantLinear(self.weight_bit, bias_bit=self.bias_bit, 
+                quant_mode=self.quant_mode, per_channel=True)
+        self.dense.set_param(nn.Linear(config.intermediate_size, config.hidden_size))
+        self.ln_input_act = QuantAct(self.ln_input_bit, quant_mode=self.quant_mode)
+        self.LayerNorm = IntLayerNorm(self.ln_output_bit, quant_mode=self.quant_mode)
+        self.LayerNorm.set_param(nn.LayerNorm(config.hidden_size, eps=config.layer_norm_eps))
+        self.output_activation = QuantAct(self.act_bit, quant_mode=self.quant_mode) 
+        self.dropout = nn.Dropout(config.hidden_dropout_prob)
+
+    def forward(self, hidden_states, hidden_states_scaling_factor, 
+               input_tensor, input_tensor_scaling_factor):
+        hidden_states, hidden_states_scaling_factor= \
+                self.dense(hidden_states, hidden_states_scaling_factor)
+        hidden_states = self.dropout(hidden_states)
+        hidden_states, hidden_states_scaling_factor = self.ln_input_act(
+                hidden_states, hidden_states_scaling_factor,
+                identity=input_tensor,
+                identity_scaling_factor=input_tensor_scaling_factor)
+        hidden_states, hidden_states_scaling_factor = \
+                self.LayerNorm(hidden_states, hidden_states_scaling_factor)
+
+        hidden_states, hidden_states_scaling_factor = \
+                self.output_activation(hidden_states, hidden_states_scaling_factor) 
+        return hidden_states, hidden_states_scaling_factor
+
+
+# Referred from transformers.models.bert.modeling_bert.BertLayer
+class IBertLayer(nn.Module):
+    def __init__(self, config):
+        super().__init__()
+        self.quant_mode = config.quant_mode
+        self.act_bit = 8
+
+        self.seq_len_dim = 1
+        self.chunk_size_feed_forward = config.chunk_size_feed_forward
+        self.is_decoder = config.is_decoder
+        self.add_cross_attention = config.add_cross_attention
+        assert self.chunk_size_feed_forward == 0
+        assert not self.is_decoder
+        assert not self.add_cross_attention
+
+        self.attention = IBertAttention(config)
+        self.intermediate = IBertIntermediate(config)
+        self.output = IBertOutput(config)
+
+        self.pre_intermediate_act = QuantAct(self.act_bit, quant_mode=self.quant_mode)
+        self.pre_output_act = QuantAct(self.act_bit, quant_mode=self.quant_mode)
+
+    def forward(
+        self,
+        hidden_states,
+        hidden_states_scaling_factor,
+        attention_mask=None,
+        head_mask=None,
+        encoder_hidden_states=None,
+        encoder_attention_mask=None,
+        past_key_value=None,
+        output_attentions=False,
+    ):
+        assert past_key_value is None
```

Comment:
Could we add messages to the assertions so that we know what's the issue if it crashes?